### ISSUE TITLE
Inconsistent behavior on CloudFormation Template regarding BinaryMediaType

### ISSUE THREAD

**Issue by homanleest** [awslabs/serverless-application-model#1687]
Not sure is that the right channel for this but basically we hit a really inconsistent issue when deploying our APIGateway  through CloudFormation template 

**Description:**

When we set the BinaryMediaTypes option in CFT AWS::ApiGateway::RestApi resource, it have unpredictable behavior after the stack is build/updated. it will have some media-type missing. For example, if we put setting like this:    
BinaryMediaTypes:
        - 'image/png'
        - 'image/jpeg'
        - 'multipart/form-data'
        - 'application/pdf'
In API Gateway after the stack has successfully rebuild with this template change, it will only have 0-4 of above media-type set in APIGateway. The most recent one is the image/jpeg missing from the APIGW dashboard

**Steps to reproduce the issue:**
1. Run the CF stack update by running with the template with above setting:
  ApiGateway:
    Type: 'AWS::ApiGateway::RestApi'
    Description: The API Gateway proxying requests to the API server
    Properties:
      Description: !Sub 'A RESTful API for a ${ApplicationName}'
      Body: 
        [swagger thing]
      BinaryMediaTypes:
        - [x number of items]
2. Wait for the build is finish in CloudFormation
3. Go to API Gatwway dashboard, find the API you just updated by CF, go to "Settings" of that API. observe the "Binary Media Types" section

**Observed result:**
Have zero to x number of items under that section. Randomly and have no true logic I can think off

**Expected result:**
Have the x number of items set as in the CloudFormation template

**Comment by jfuss** [comment]
@homanleest Thanks for the issue but this seems like you are using the basic API Gateway resource (AWS::ApiGateway::RestApi) not SAM's (AWS::Serverless::API). It seems like in SAM we add Binary Media types to both the AWS::ApiGateway::RestApi we generate (like you have it) and to the swagger document (https://github.com/awslabs/serverless-application-model/blob/8b3ea4f733ba9d90caf5a97504e6f9024c29c0cf/samtranslator/model/api/api_generator.py#L142 and https://github.com/awslabs/serverless-application-model/blob/8b3ea4f733ba9d90caf5a97504e6f9024c29c0cf/samtranslator/model/api/api_generator.py#L171). I would suggest trying that to see if that solves the problem for you. 

I reached out to the service team as well. Will respond back if there are more details. 

Closing this as this does not relate to SAM but instead the vanilla CloudFormation Resource for API Gateway.

**Comment by homanleest** [comment]
Tried and it works! thanks for the suggestion. We spent couple days and multiple dev to look into this issue.

That need to be written to the CFT documentation... The odd thing is not only it doesn't work by just setting swagger doc only but also doesn't work by just setting the CFT setting. we need to set BOTH to have that to work. Put it here so that the next one googling this or even AWS tech support can see that. 

Hope ST can fix that one day... either deprecate CFT setting to force us to use swagger setting or either way should work if and document that when it will deprecate the CFT setting .
